annotations.md

```diff
@@ -0,0 +1,230 @@
+## Binding Metadata in Annotations
+
+ During a binding operation, annotations from relevant Kubernetes resources are extracted to gather information about what is interesting for binding. This information is eventually used to bind the application with the backing service by populating the binding secret.
+
+### Requirements for specifying binding information in a backing service CRD / Kubernetes resource
+
+1. Extract a string from the Kubernetes resource.
+2. Extract a string from the Kubernetes resource, and map it to custom name in the binding secret.
+3. Extract an entire configmap/secret from the Kubernetes resource.
+4. Extract a specific field from the configmap/secret from the Kubernetes resource, and bind it as an environment variable.
+5. Extract a specific field from the configmap/secret from the Kubernetes resource and and bind it as a volume mount.
+6. Extract a specific field from the configmap/secret from the Kubernetes resource and map it to different name in the binding secret.
+7. Extract a “slice of maps” from the Kubernetes resource and generate multiple fields in the binding secret.
+8. Extract a "slice of strings" from a Kubernetes resource and indicate the content in a specific index in the slice to be relevant for binding.
+
+
+### Data model : Building blocks for expressing binding information
+
+* `path`: A template representation of the path to the element in the Kubernetes resource.
```

Comment:
We should probably specify here that this could be a [JSONPath](https://kubernetes.io/docs/reference/kubectl/jsonpath/) or [GO templates](https://golang.org/pkg/text/template/).

Reply:
Done.